### ISSUE TITLE
`mismatched output from interactive.diffFilter` when adding the existence of new files

### ISSUE THREAD

**Issue by llakala** [so-fancy/diff-so-fancy#498]
> I've been testing the `--patch` mode for a couple of days with good results in almost all cases. The only problem I've found is when you do the following:
> 
>     1. Create a new file with content
> 
>     2. Use `git add --intent-to-add /path/to/file`
> 
>     3. Now `git add -p` fails with the error below
> 
> 
> ```
> fatal: mismatched output from interactive.diffFilter
> hint: Your filter must maintain a one-to-one correspondence
> hint: between its input and output lines.
_Originally posted by @wvega-godaddy in https://github.com/so-fancy/diff-so-fancy/issues/35#issuecomment-749565403_
            
Having the same issue here, on Git 2.47.0 and diff-so-fancy 1.4.4. I use `--intent-to-add` all the time in NixOS, because any untracked files are ignored by the system. Would love to see this fixed.

**Comment by OJFord** [comment]
And you do have `git config --global interactive.diffFilter 'diff-so-fancy --patch'` set?

**Comment by llakala** [comment]
> And you do have `git config --global interactive.diffFilter 'diff-so-fancy --patch'` set?

Yes, see this [Asciinema](https://asciinema.org/a/S5Hrdy0SHIR8mTNKT0T1LFHZf) for a demonstration of how to replicate the issue.

**Comment by tdewolff** [comment]
Same thing happens for me, diff-so-fancy v1.4.4 and git v2.51.2. I'm adding a new (binary: PDF) file and getting the same error. I have set `git config --global interactive.diffFilter 'diff-so-fancy --patch'`.

**Comment by scottchiefbaker** [comment]
Thank you for the Asciinema, that is **very** helpful. What I'm seeing is that when we use `--patch` it adds an extra "\n" to the output to maintain the input:output line ratio on textual diffs. On new file adds we do **not** need those extra lines. 

In other words, if we disable patch mode in d-s-f and follow the steps in the Asciinema the issue goes away. We need some way to know that we're adding a binary file and NOT to output the extra lines.

**Comment by scottchiefbaker** [comment]
This should be fixed in the latest v1.4.6 release. Please test and let me know.

**Comment by FatBoyXPC** [comment]
I have this exact issue with git 2.54.0 and dsf 1.4.10, also on nixos.

```
$ git --version && diff-so-fancy --version
git version 2.54.0
Diff-so-fancy: https://github.com/so-fancy/diff-so-fancy
Version      : 1.4.10
```
